zenoh/src/api/builders/querier.rs

```diff
@@ -0,0 +1,450 @@
+    ///     .unwrap();
+    /// let mut n = 0;
+    /// let _ = querier
+    ///     .get()
+    ///     .callback_mut(move |reply| {n += 1;})
+    ///     .await
+    ///     .unwrap();
+    /// # }
+    /// ```
+    #[inline]
+    pub fn callback_mut<F>(self, callback: F) -> QuerierGetBuilder<'a, 'b, Callback<Reply>>
+    where
+        F: FnMut(Reply) + Send + Sync + 'static,
+    {
+        self.callback(locked(callback))
+    }
+
+    /// Receive the replies for this query with a [`Handler`](crate::handlers::IntoHandler).
+    ///
+    /// # Examples
+    /// ```
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// use zenoh::{query::{ConsolidationMode, QueryTarget}};
+    ///
+    /// let session = zenoh::open(zenoh::Config::default()).await.unwrap();
+    /// let querier = session.declare_querier("key/expression")
+    ///     .target(QueryTarget::All)
+    ///     .consolidation(ConsolidationMode::None)
+    ///     .await
+    ///     .unwrap();
+    /// let replies = querier
+    ///     .get()
+    ///     .with(flume::bounded(32))
+    ///     .await
+    ///     .unwrap();
+    /// while let Ok(reply) = replies.recv_async().await {
+    ///     println!("Received {:?}", reply.result());
+    /// }
+    /// # }
+    /// ```
+    #[inline]
+    pub fn with<Handler>(self, handler: Handler) -> QuerierGetBuilder<'a, 'b, Handler>
+    where
+        Handler: IntoHandler<Reply>,
+    {
+        let QuerierGetBuilder {
+            querier,
+            parameters,
+            value,
+            attachment,
+            #[cfg(feature = "unstable")]
+            source_info,
+            handler: _,
+        } = self;
+        QuerierGetBuilder {
+            querier,
+            parameters,
+            value,
+            attachment,
+            #[cfg(feature = "unstable")]
+            source_info,
+            handler,
+        }
+    }
+}
+impl<'a, 'b, Handler> QuerierGetBuilder<'a, 'b, Handler> {
+    /// Set the query payload.
+    #[inline]
+    pub fn payload<IntoZBytes>(mut self, payload: IntoZBytes) -> Self
+    where
+        IntoZBytes: Into<ZBytes>,
+    {
+        let mut value = self.value.unwrap_or_default();
+        value.0 = payload.into();
+        self.value = Some(value);
+        self
+    }
+
+    /// Set the query selector parameters.
+    #[inline]
+    pub fn parameters<P>(mut self, parameters: P) -> Self
+    where
+        P: Into<Parameters<'b>>,
+    {
+        self.parameters = parameters.into();
+        self
+    }
+}
+
+impl<Handler> Resolvable for QuerierGetBuilder<'_, '_, Handler>
+where
+    Handler: IntoHandler<Reply> + Send,
+    Handler::Handler: Send,
+{
+    type To = ZResult<Handler::Handler>;
+}
+
+impl<Handler> Wait for QuerierGetBuilder<'_, '_, Handler>
+where
+    Handler: IntoHandler<Reply> + Send,
+    Handler::Handler: Send,
+{
+    fn wait(self) -> <Self as Resolvable>::To {
+        let (callback, receiver) = self.handler.into_handler();
+
+        #[allow(unused_mut)]
```

Comment:
Add a comment when the `unused_mut` occurs.

Reply:
fixed